R/utilities-goodness-of-fit.R

```diff
@@ -553,6 +572,65 @@ plotMeanTimeProfile <- function(simulatedData,
   return(timeProfilePlot)
 }
 
+#' @title plotMeanTimeProfileLog
+#' @description Plot time profile for mean model workflow
+#' @param simulatedData data.frame of simulated data
+#' @param observedData data.frame of observed data
+#' @param lloqData data.frame of lloq data
+#' @param metaData meta data on `data`
+#' @param dataMapping A list mapping `x`, `y` and `group` from datasets
+#' @param plotConfiguration `PlotConfiguration` class object from `tlf` library
+#' @return ggplot object of time profile for mean model workflow
+#' @export
+#' @import tlf
+plotMeanTimeProfileLog <- function(simulatedData,
+                                   observedData = NULL,
+                                   lloqData = NULL,
+                                   metaData = NULL,
+                                   dataMapping = NULL,
+                                   plotConfiguration = NULL) {
+  # Remove 0 values from simulated and observed data
+  # TODO: use dplyr to refactor the selection
+  simulatedData <- removeNegativeValues(simulatedData, dataMapping$y)
+  observedData <- removeNegativeValues(observedData, dataMapping$y)
+  lloqData <- removeNegativeValues(lloqData, dataMapping$y)
```

Comment:
There are some problems with this solution. 
**But I would suggest to address them in the separate PR in order not to block this PR for even longer.**
(just created an issue for this: https://github.com/Open-Systems-Pharmacology/OSPSuite.ReportingEngine/issues/942)

I have already created the issue for this in the ospuite-R (because residuals are calculated there as well):
https://github.com/Open-Systems-Pharmacology/OSPSuite-R/issues/1091

In short:
* Removing <=0 values (especially observed values) is not correct. E.g. if we have `observed = 0` and `simulated >>0`, we just ignore a significant residual in such a case.
* Very small positive values are problematic for log residuals as well. E.g. having `observed = 1e-100` and `simulated = 1e-200` would produce a huge log residual, but in fact the difference is irrelevant - both values are numerical zero.

For this, PK-Sim/MoBi introduces the `LogSafe `function, which is used for calculating log residuals and defined as:
`LogSafe(Y) = Log(Max(Y, LogSafeEpsilon))` (with LogSafeEpsilon currently defined as 1E-20)

And Lin residuals are calculated with all the values AS IS